keyboards/ikeygai/albatross/albt_s/albt_s.c

```diff
@@ -0,0 +1,23 @@
+/* Copyright 2023 Gondolindrim <gondolindrim@acheronproject.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    led_state.caps_lock ? backlight_enable() : backlight_disable() ; 
+    return res;
+}
```

Comment:
This implementation is incorrect in 2 ways
1. It will write to eeprom on every time the caps lock state changes
2. It does not correctly reflect the return of `led_update_user` when applying the keyboard level logic

Also it doesnt seem to add any value over the existing `BACKLIGHT_CAPS_LOCK` option.

Reply:
`BACKLIGHT_CAPS_LOCK` does not work. It makes the caps lock LED turn on when caps lock is off, and the LED turn off when caps is on.

Reply:
`on_state: 0` does a very interesting thing in that the LED brightness does not work. Say I set the brightness to 20%, so when caps is on the brightness is 20% but it goes to 80% when off. The only way the caps lock works is at full brightness.

Reply:
Hm, with backlight off and `on_state: 1` (testing with N-channel MOSFET), it appears to work fine - it is meant to turn the backlight on at full brightness when caps lock is on.
However with the backlight at some non-zero brightness level it appears to be inverted:

```c
    // Use backlight as Caps Lock indicator
    uint8_t bl_toggle_lvl = 0;

    if (led_state.caps_lock && !backlight_config.enable) {
        // Turning Caps Lock ON and backlight is disabled in config
        // Toggling backlight to the brightest level
        bl_toggle_lvl = BACKLIGHT_LEVELS;
    } else if (!led_state.caps_lock && backlight_config.enable) {
        // Turning Caps Lock OFF and backlight is enabled in config
        // Toggling backlight and restoring config level
        bl_toggle_lvl = backlight_config.level;
    }

    // Set level without modify backlight_config to keep ability to restore state
    backlight_set(bl_toggle_lvl);
```

Reply:
This is possibly intentional considering if you have backlight *and* caps lock on, then how do you know what the caps lock state is?
It seems someone else noticed already: https://github.com/qmk/qmk_firmware/pull/4769#issuecomment-647191928

Reply:
I will be honest that, at a first glance, this does not make sense to me. In my head what would be better is: if backlight is defined and `as_caps_lock` too, then the backlight should be always at the configured level if both are on... Something like this:

```
    uint8_t bl_toggle_lvl = 0;

    if (led_state.caps_lock && backlight_config.enable) bl_toggle_lvl = backlight_config.level;
    } else bl_toggle_lvl = 0;
 
    backlight_set(bl_toggle_lvl);
```

Or to save code and the variable:

```
    (led_state.caps_lock && backlight_config.enable) ? backlight_set(backlight_config.level) : backlight_set(0);
```

Reply:
I have just confirmed that this latest option works. This is `albt_s.c`:

```c
#include "backlight.h"
extern backlight_config_t backlight_config;

bool led_update_kb(led_t led_state) {
    bool res = led_update_user(led_state);
    (led_state.caps_lock && backlight_config.enable) ? backlight_set(backlight_config.level) : backlight_set(0);
    return res;
}
```

Reply:
So essentially:
```
backlight off, caps lock off => off
backlight off, caps lock on => off
backlight on, caps lock off => off
backlight on, caps lock on => on
```
In other words `backlight & caps_lock`.
Which I think might be slightly confusing to users trying to adjust the backlight level and having nothing happen, until they turn on caps lock. But this could be easily called out in the docs, after all, this config is opt-in and should not be enabled by default at either keymap or keyboard level...

There are currently five boards in the repo with `as_caps_lock: true`:
```
mode\m75s
mokey\ginkgo65
mokey\ginkgo65hot
sentraq\s65_x
smithrune\iron180v2\v2s
```

Reply:
My reasoning here is: if `BACKLIGHT_CAPS_LOCK` is defined, since it's not a default, this means that the designer specifically wants the backlight to be on only when caps lock is also on, exactly how you said, `backlight & caps_lock`. I don't know about the other PCBs, but for M75S, Iron180V2 and this PCB, only the caps lock key has an LED, so if the user is soldering an LED on caps lock that pretty much means that the LED is an indicator. If a PCB I'm designing has LEDs on every single switch,  what I do is I add a second MOSFET to caps lock so that allows the PCB to control caps lock individually as an indicator while the rest of the PCB LEDs work at the backlight level.

Reply:
I pushed this in [20308ac](https://github.com/qmk/qmk_firmware/pull/22941/commits/20308ac310f044cca9c742aa991c0ef7e486a79a). It seems to be a good compromise between the points @zvecr raised and the discussion me and @fauxpark had.